ios/RCTMapboxGL/RCTMapboxGL.m

```diff
@@ -117,24 +116,19 @@ - (void)layoutSubviews
 
 - (void)setAnnotations:(NSMutableArray *)annotations
 {
-    _newAnnotations = annotations;
-    [self performSelector:@selector(updateAnnotations) withObject:nil afterDelay:0.5];
+    [self performSelector:@selector(updateAnnotations:) withObject:annotations afterDelay:0.5];
 }
 
-- (void)updateAnnotations {
-    if (_newAnnotations) {
-        if (_annotations.count) {
-            for (int i = 0; i < [_annotations count]; i++) {
-                [_map removeAnnotations: _annotations];
-            }
-            _annotations = nil;
-        }
-
-        _annotations = _newAnnotations;
-        for (int i = 0; i < [_newAnnotations count]; i++) {
-            [_map addAnnotation:_newAnnotations[i]];
+- (void)updateAnnotations:(NSMutableArray *) annotations {
+    for (int i = 0; i < [annotations count]; i++) {
```

Comment:
It would potentially be a lot more efficient to write `for (RCTMGLAnnotation *annotation in annotations)` instead of repeated calls to `-objectAtIndex:`.